src/se/vidstige/jadb/HostConnectToRemoteTcpDevice.java

```diff
@@ -0,0 +1,76 @@
+package se.vidstige.jadb;
+
+import se.vidstige.jadb.entities.TcpAddressEntity;
+
+import java.io.IOException;
+
+class HostConnectToRemoteTcpDevice {
+    private final Transport transport;
+    private final ResponseValidator responseValidator;
+
+    HostConnectToRemoteTcpDevice(Transport transport) {
+        this.transport = transport;
+        this.responseValidator = new ResponseValidatorImp();
+    }
+
+    //Visible for testing
+    HostConnectToRemoteTcpDevice(Transport transport, ResponseValidator responseValidator) {
+        this.transport = transport;
+        this.responseValidator = responseValidator;
+    }
+
+    TcpAddressEntity connect(TcpAddressEntity tcpAddressEntity)
+            throws IOException, JadbException, ConnectionToRemoteDeviceException {
+        transport.send(String.format("host:connect:%s:%d", tcpAddressEntity.getHost(), tcpAddressEntity.getPort()));
```

Comment:
@vidstige I think that JadbConnection is growing  - and I think it is better to delegate implementation of concrete command to it's own class.
It is also allows to test these classes isolated - for example in `HostConnectToRemoteTcpDeviceTest` and `HostDisconnectFromRemoteTcpDeviceTest` uses:

```
 Transport transport = mock(Transport.class);
 doThrow(new JadbException("Fake exception")).when(transport).verifyResponse();
```

to simulate error